source/common/http/async_client_impl.h

```diff
@@ -164,6 +164,13 @@ class AsyncStreamImpl : public AsyncClient::Stream,
       return nullptr;
     }
 
+    Router::RouteConstSharedPtr route(const Router::RouteCallback& cb,
+                                      const Http::RequestHeaderMap&, const StreamInfo::StreamInfo&,
+                                      uint64_t) const override {
+      cb(nullptr, Router::RouteEvalStatus::NoMoreRoutes);
+      return nullptr;
```

Comment:
I would just NOT_IMPLEMENTED this variant and the one below. It won't ever get called.